proto/worker_status.proto

```diff
@@ -0,0 +1,30 @@
+// Specification of the user facing gRPC API.
+syntax = "proto3";
+package status;
+
+service StatusApi {
+    // Returns the status of the worker.
+    rpc Status(StatusRequest) returns (StatusResponse) {}
+}
+
+// Request message for the status of the worker.
+message StatusRequest {}
+
+// Response message containing the status of the worker.
+message StatusResponse {
+    // Whether the worker is ready to generate proofs.
+    bool ready = 1;
+    // The version of the worker.
+    string version = 2;
+    // The supported proof types.
+    SupportedProofTypes supported_proof_types = 3;
```

Comment:
Should this not be an enum? I'm assuming here that a worker can be started only in one "mode" (i.e., to generate either transaction, batch, or block proofs).

Reply:
What I did was allow multiple proof types on the same worker (and also on the proxy side of this request), ensuring that both sides have the same proof types enabled. So, by setting a single "mode" on a worker, we achieve that same goal, but still allow anyone who wants to use the worker at their own risk for general purposes.

Although if we want to make a hard requirement of this, we could change it to an enum.